src/urllib3/connectionpool.py

```diff
@@ -313,7 +319,7 @@ def _validate_conn(self, conn):
         """
         pass
 
-    def _prepare_proxy(self, conn):
+    def _prepare_proxy(self, conn, destination_scheme):
```

Comment:
Comment that must have gotten lost when I switched from mobile to PC:

Had a thought on being able to detect whether a `conn` is connected to HTTPS then we wouldn't need `destination_scheme` I believe? I saw many libraries defining `_prepare_proxy()` on GitHub, even though it's a "private" method we'd be breaking those users if we modified the signature like this so would be best if we could avoid doing so.

Initial thoughts: checking `hasattr(conn, "ssl_context")`, `isinstance(conn, HTTPSConnection)`? Since we don't support any of this unless we're using `HTTPSConnection` that might be enough?

Reply:
Good point. After reviewing about this for a while, we don't really need to modify the ``prepare_proxy`` method signature, just call it when we really need to do the HTTP CONNECT. I made some simplifications, let me know what you think!